networkx/generators/small.py

```diff
@@ -768,7 +771,55 @@ def petersen_graph(create_using=None):
     G.name = "Petersen Graph"
     return G
 
+@nx._dispatchable(graphs=None, returns_graph=True)
+def generalized_petersen_graph(n, k, create_using=None):
+    """
+    Returns the generalized Petersen graph GP(n,k).
+
+    The Generalized Peterson graph consists of an outer cycle of n nodes
+    connected to an inner circulant graph of n nodes, where nodes in the
+    inner circulant are connected to their kth nearest neighbor [1]_. A
+    generalized Petersen graph is cubic with 2n nodes and 3n edges. 
+    
+    Some well known graphs are examples of generalized Petersen graphs such
+    as the Desargues graph GP(10,3) and dodecahedron graph GP(10,2).
+
+    Parameters
+    ----------
+    n : int
+       Number of nodes in the outer cycle and inner circulant. n must be greater or equal to 3.
+       
+    k : int
+       Neighbor to connect in the inner circulant. 1 <= k <= floor( (n-1)/2 )
+
+    create_using : NetworkX graph constructor, optional (default=nx.Graph)
+       Graph type to create. If graph instance, then cleared before populated.
+
+    Returns
+    -------
+    G : networkx Graph
+        Generalized Petersen graph n k
+
+    References
+    ----------
+    .. [1] https://mathworld.wolfram.com/GeneralizedPetersenGraph.html
+    """
+    if n <= 2:
+        raise NetworkXError("n should be greater than or equal to 3")
+    if k <= 0 or k > int((n-1)/2):
+        raise NetworkXError("k should be greater than or equal to 1 and smaller than floor((n-1)/2)")
 
+
+    G = nx.Graph()
+    for i in range(n):
+        G.add_edge(i, (i+1)%n)
```

Comment:
Can you run the linting tools on this code?
Install the package pre-commit. Then run `pre-commit install` from the local repo.
And then you can run `pre-commit run --files small.py` (or prepend the path to `small.py` if you need to).

The linting tools will fix the white-space and other python style ideas to match the other code.